server/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -259,6 +263,16 @@ SegmentIdWithShardSpec allocatePendingSegment(
    * @throws IllegalArgumentException if startMetadata and endMetadata are not either both null or both non-null
    * @throws RuntimeException         if the state of metadata storage after this call is unknown
    */
+  SegmentPublishResult announceHistoricalSegments(
+      Set<DataSegment> segments,
+      Set<DataSegment> segmentsToDrop,
+      @Nullable DataSourceMetadata startMetadata,
+      @Nullable DataSourceMetadata endMetadata,
+      @Nullable Map<DataSegment, TaskLockInfo> segmentLockMap,
+      @Nullable Set<TaskLockInfo> taskLockInfos,
+      boolean append
+  ) throws IOException;
+
```

Comment:
This is running the risk of changing the logic of how the other locks operate.  I'd rather not take that risk.  Instead of trying to reuse this method.  Please create 2 new methods:

```
commitReplaceSegments()
commitAppendSegments()
```

And have the new locks use those methods instead.  If you do that, then the code for the DB interactions should all be new method implementations without any interleaved changes with old logic, which is an easier way to validate that old logic goes unchanged.

Reply:
Done